packages/snaps-controllers/src/snaps/endowments/name-lookup.ts

```diff
@@ -94,21 +169,31 @@ function validateCaveat(caveat: Caveat<string, any>): void {
 export function getNameLookupCaveatMapper(
   value: Json,
 ): Pick<PermissionConstraint, 'caveats'> {
-  if (
-    !value ||
-    !Array.isArray(value) ||
-    (Array.isArray(value) && value.length === 0)
-  ) {
+  if (!value || !isObject(value) || Object.keys(value).length === 0) {
     return { caveats: null };
   }
-  return {
-    caveats: [
-      {
-        type: SnapCaveatType.ChainIds,
-        value,
-      },
-    ],
-  };
+
+  const caveats = [];
+
+  if (value.chains) {
+    caveats.push({
+      type: SnapCaveatType.ChainIds,
+      value: value.chains,
+    });
+  }
+
+  if (value.matchers) {
+    caveats.push({
```

Comment:
Do we need this if we have `getMatchersCaveat`?

I guess we need to decide if it is supposed to be two caveats or one? It could be a shared caveat with both values in it, if we want it to

Reply:
I'd rather it be two caveats, the `chainIds` caveat can be used in other permissions as well.

Reply:
Cool, then this getter shouldn't care about `matchers`